src/overlays/actors/ovl_En_Syateki_Man/z_en_syateki_man.c

```diff
@@ -41,73 +56,1252 @@ const ActorInit En_Syateki_Man_InitVars = {
+                break;
+
+            case 1029:
+            case 1030:
+            case 1031:
+                globalCtx->msgCtx.unk11F22 = 0x43;
+                globalCtx->msgCtx.unk12023 = 4;
+                player->actor.freezeTimer = 0;
+                gSaveContext.minigameState = 3;
+                this->actionFunc = func_809C7D14;
+                func_809C7D14(this, globalCtx);
+                break;
+        }
+    }
+}
+
+void func_809C7990(EnSyatekiMan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (player->stateFlags1 & 0x20) {
+        player->stateFlags1 |= 0x20;
+    }
+
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 2:
+            this->actionFunc = func_809C72D8;
+            this->unk_26A = 0;
+            break;
+
+        case 4:
+            func_809C7380(this, globalCtx);
+            break;
+
+        case 5:
+            func_809C7620(this, globalCtx);
+            break;
+
+        case 6:
+            if (func_80147624(globalCtx)) {
+                gSaveContext.weekEventReg[63] &= (u8)~1;
+                gSaveContext.weekEventReg[63] &= (u8)~2;
+                player->stateFlags1 &= ~0x20;
+                this->actionFunc = func_809C72D8;
+                this->unk_26A = 0;
+            }
+            break;
+
+        case 0:
+        case 1:
+        case 3:
+        case 7:
+        case 8:
+        case 9:
+        case 10:
+            break;
+    }
+}
+
+void func_809C7A90(EnSyatekiMan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        if (!(gSaveContext.weekEventReg[59] & 0x10)) {
+            gSaveContext.weekEventReg[59] |= 0x10;
+        } else if (!(gSaveContext.weekEventReg[32] & 2) && (this->unk_280 >= 0x884)) {
+            gSaveContext.weekEventReg[32] |= 2;
+        }
+        this->actor.parent = NULL;
+        this->actionFunc = func_809C7C14;
+    } else {
+        if ((CUR_UPG_VALUE(UPG_QUIVER) < 3) && !(gSaveContext.weekEventReg[59] & 0x10)) {
+            func_800B8A1C(&this->actor, globalCtx, GI_QUIVER_30 + CUR_UPG_VALUE(UPG_QUIVER), 500.0f, 100.0f);
+        } else if (this->unk_280 < 0x884) {
+            func_800B8A1C(&this->actor, globalCtx, GI_RUPEE_RED, 500.0f, 100.0f);
+        } else if (!(gSaveContext.weekEventReg[32] & 2)) {
+            func_800B8A1C(&this->actor, globalCtx, GI_HEART_PIECE, 500.0f, 100.0f);
+        } else {
+            func_800B8A1C(&this->actor, globalCtx, GI_RUPEE_PURPLE, 500.0f, 100.0f);
+        }
+
+        player->actor.shape.rot.y = -0x8000;
+        player->actor.velocity.z = 0.0f;
+        player->actor.velocity.x = 0.0f;
+        player->actor.world.rot.y = player->actor.shape.rot.y;
+    }
+}
+
+void func_809C7C14(EnSyatekiMan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        if ((CURRENT_DAY == 3) && (gSaveContext.time > 0x8000)) {
```

Comment:
```suggestion
        if ((CURRENT_DAY == 3) && (gSaveContext.time > CLOCK_TIME(12, 00))) {
```
See tools/timeconv.py